src/main/java/com/dokany/java/DokanyMount.java

```diff
@@ -64,70 +62,79 @@ public long getVersion() {
 
 	/**
 	 * Calls {@link NativeMethods#DokanMain(DeviceOptions, DokanyOperations)} in a thread of the Common ForkJoin Pool.
-	 * Adds to the JVM a {@link java.lang.Runtime#addShutdownHook(Thread)} which calls {@link #close()}
+	 * No-op if this object is already mounted.
+	 * <p>
+	 * Additionally a shutdown hook invoking {@link #close()} is registered to the JVM.
 	 */
-	public synchronized void mount() throws DokanyException {
-		this.mount(ForkJoinPool.commonPool());
+	public void mount() throws DokanyException {
+		this.mount(ForkJoinPool.commonPool(), () -> {});
 	}
 
+
 	/**
 	 * Calls {@link NativeMethods#DokanMain(DeviceOptions, DokanyOperations)} on the given executor.
-	 * Adds to the JVM a {@link java.lang.Runtime#addShutdownHook(Thread)} which calls {@link #close()}
+	 * No-op if this object is already mounted.
+	 * <p>
+	 * Additionally a shutdown hook invoking {@link #close()} is registered to the JVM.
+	 *
+	 * @param executor
+	 * @throws DokanyException
+	 */
+	public void mount(Executor executor) throws DokanyException {
+		this.mount(executor, () -> {});
+	}
+
+
+	/**
+	 * Calls {@link NativeMethods#DokanMain(DeviceOptions, DokanyOperations)} on the given executor and runs after mount termination a specified action.
+	 * No-op if this object is already mounted.
+	 * <p>
+	 * Additionally a shutdown hook invoking {@link #close()} is registered to the JVM.
 	 *
-	 * @Param Executor executor
+	 * @param executor
+	 * @param onDokanExit object with a run() method which is executed after the Dokan process exited
 	 */
-	public synchronized void mount(Executor executor) throws DokanyException {
+	public synchronized void mount(Executor executor, Runnable onDokanExit) throws DokanyException {
 		if (!isMounted) {
+			isMounted = true;
 			try {
 				Runtime.getRuntime().addShutdownHook(new Thread(this::close));
 
 				LOG.info("Dokany API/driver version: {} / {}", getVersion(), getDriverVersion());
 				//real mount op
-				mountFuture = CompletableFuture
-						.supplyAsync(() -> NativeMethods.DokanMain(deviceOptions, new DokanyOperationsProxy(fileSystem)), executor)
-						.handle((returnVal, exception) -> {
-							setIsMounted(false);
-							if (returnVal != null && returnVal.equals(0)) {
-								return 0;
+				CompletableFuture.supplyAsync(() -> NativeMethods.DokanMain(deviceOptions, new DokanyOperationsProxy(fileSystem)), executor)
+						.whenComplete((returnVal, throwable) -> {
+							isMounted = false;
+							onDokanExit.run();
+							if (throwable != null) {
+								throw new DokanyException(throwable);
 							}
-
-							if (returnVal == null) {
-								throw new DokanyException(exception);
-							} else {
-								throw new DokanyException("Return Code " + returnVal + ": " + MountError.fromInt(returnVal).getDescription());
+							if (returnVal != null && returnVal != 0) {
```

Comment:
Can we `assert returnVal != null`? At this point either throwable or returnVal should be non-null (if I'm not mistaken).